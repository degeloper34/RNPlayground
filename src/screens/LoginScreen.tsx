import {Alert, Pressable, StyleSheet, View} from "react-native";
import {RootStackScreenProps} from "../../types";
import Constants from "expo-constants";
import * as SecureStore from "expo-secure-store";
import {memberLogin} from "../api/services/memberService";
import {
  CustomText,
  FlexView,
  Loading,
  CustomTextInput,
  CustomButton,
} from "../components/atoms";
import {useContext, useState} from "react";
import {MainContext} from "../context/mainContext";
import Colors from "../constants/Colors";

export default function LoginScreen({
  navigation,
}: RootStackScreenProps<"Login">) {
  const [username, setUsername] = useState<string>("johnd");
  const [password, setPassword] = useState<string>("m38rmF$");
  const context = useContext(MainContext);
  const version = Constants?.manifest?.version;

  const onPressLoginBtn = async () => {
    context.setAppLoading(true);
    const memberLoginResponse = await memberLogin(username, password);
    console.log("memberLoginResponse", memberLoginResponse);

    context.setAppLoading(false);

    if (memberLoginResponse) {
      await SecureStore.setItemAsync("memberToken", memberLoginResponse.token);
      navigation.navigate("Root");
    }
  };

  const onPressWorkInProgress = () => {
    Alert.alert("Work in progress");
  };

  const onChangeUsername = (text: string) => {
    setUsername(text);
  };

  const onChangePassword = (text: string) => {
    setPassword(text);
  };

  const {
    container,
    panelLogin,
    marginTop,
    btnForgotPassword,
    btnSignUp,
    txtVersion,
  } = styles;

  return (
    <View style={container}>
      <Loading isLoadingActive={context?.appLoading} />
      <FlexView />
      <CustomText
        type={"bold"}
        fontSize={24}
        text={"Hello,"}
        textColor={Colors.mainGrey}
      />
      <CustomText
        type={"bold"}
        fontSize={24}
        text={"Login now"}
        textColor={Colors.mainYellow}
      />

      <View style={panelLogin}>
        <FlexView flex={6}>
          <CustomTextInput
            placeholder="username"
            value={username}
            onChangeText={onChangeUsername}
            maxLength={30}
          />

          <View style={marginTop} />
          <CustomTextInput
            placeholder="password"
            value={password}
            secureTextEntry
            onChangeText={onChangePassword}
            maxLength={30}
          />

          <Pressable onPress={onPressWorkInProgress} style={btnForgotPassword}>
            <FlexView />
            <CustomText
              type={"medium"}
              fontSize={12}
              text={"Forgot password?"}
              textColor={Colors.mainYellow}
            />
          </Pressable>

          <FlexView flex={2} />
        </FlexView>

        <CustomButton text={"Login"} onPress={onPressLoginBtn} />
      </View>
      <Pressable onPress={onPressWorkInProgress} style={btnSignUp}>
        <CustomText
          type={"medium"}
          fontSize={12}
          text={"Don't have an account?"}
          textColor={"white"}
        />
        <CustomText
          type={"medium"}
          fontSize={12}
          text={" Sign up"}
          textColor={Colors.mainYellow}
        />
      </Pressable>

      <View style={txtVersion}>
        <CustomText
          type={"medium"}
          fontSize={10}
          text={"v" + version}
          textColor={Colors.mainGrey}
        />
      </View>

      <FlexView />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    padding: 48,
    backgroundColor: Colors.mainNight,
  },
  btnForgotPassword: {
    flexDirection: "row",
    marginTop: 12,
  },
  btnSignUp: {
    flexDirection: "row",
    marginVertical: 24,
    justifyContent: "center",
  },
  txtVersion: {
    alignItems: "center",
    marginTop: 36,
  },
  panelLogin: {
    justifyContent: "center",
    borderRadius: 8,
    flex: 1.5,
    marginTop: 24,
  },
  marginTop: {
    marginTop: 12,
  },
});
