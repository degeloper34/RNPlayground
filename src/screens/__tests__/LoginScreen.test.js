import renderer from "react-test-renderer";
import React from "react";
import LoginScreen from "../LoginScreen";

describe("LoginScreen", () => {
  it("should render without issues", () => {
    const navigate = jest.fn();
    const screen = renderer
      .create(
        <LoginScreen
          navigation={{
            navigate,
          }}
        />
      )
      .toJSON();

    expect(screen).toMatchSnapshot();
  });
});
