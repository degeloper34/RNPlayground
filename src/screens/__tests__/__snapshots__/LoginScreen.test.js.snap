// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen should render without issues 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#21325E",
      "flex": 1,
      "justifyContent": "center",
      "padding": 48,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <Text
    allowFontScaling={false}
    style={
      Array [
        undefined,
        Object {
          "color": "#F0F0F0",
          "fontFamily": "sora-bold",
          "fontSize": 24,
          "textDecorationLine": "none",
        },
      ]
    }
  >
    Hello,
  </Text>
  <Text
    allowFontScaling={false}
    style={
      Array [
        undefined,
        Object {
          "color": "#F1D00A",
          "fontFamily": "sora-bold",
          "fontSize": 24,
          "textDecorationLine": "none",
        },
      ]
    }
  >
    Login now
  </Text>
  <View
    style={
      Object {
        "borderRadius": 8,
        "flex": 1.5,
        "justifyContent": "center",
        "marginTop": 24,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 6,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 8,
              "justifyContent": "center",
              "paddingHorizontal": 6,
              "paddingVertical": 12,
            },
            Object {
              "backgroundColor": "#F0F0F0",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          maxLength={30}
          onChangeText={[Function]}
          placeholder="username"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#21325E",
            }
          }
          underlineColorAndroid="transparent"
          value="johnd"
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "borderRadius": 8,
              "justifyContent": "center",
              "paddingHorizontal": 6,
              "paddingVertical": 12,
            },
            Object {
              "backgroundColor": "#F0F0F0",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          maxLength={30}
          onChangeText={[Function]}
          placeholder="password"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "color": "#21325E",
            }
          }
          underlineColorAndroid="transparent"
          value="m38rmF$"
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 12,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        />
        <Text
          allowFontScaling={false}
          style={
            Array [
              undefined,
              Object {
                "color": "#F1D00A",
                "fontFamily": "sora-medium",
                "fontSize": 12,
                "textDecorationLine": "none",
              },
            ]
          }
        >
          Forgot password?
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 2,
          }
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F1D00A",
          "borderRadius": 30,
          "justifyContent": "center",
          "padding": 12,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            undefined,
            Object {
              "color": "#21325E",
              "fontFamily": "sora-bold",
              "fontSize": 14,
              "textDecorationLine": "none",
            },
          ]
        }
      >
        Login
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginVertical": 24,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          undefined,
          Object {
            "color": "white",
            "fontFamily": "sora-medium",
            "fontSize": 12,
            "textDecorationLine": "none",
          },
        ]
      }
    >
      Don't have an account?
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          undefined,
          Object {
            "color": "#F1D00A",
            "fontFamily": "sora-medium",
            "fontSize": 12,
            "textDecorationLine": "none",
          },
        ]
      }
    >
       Sign up
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 36,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          undefined,
          Object {
            "color": "#F0F0F0",
            "fontFamily": "sora-medium",
            "fontSize": 10,
            "textDecorationLine": "none",
          },
        ]
      }
    >
      vundefined
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
